File: 9000-9999/9639 - Dinonun böyük massivi/solution.cpp
#include <iostream>
#include <vector>
#include <bits/stdc++.h>
using namespace std;

struct COMPARSION{
  inline bool operator() (const vector<long long> p1,const vector<long long> p2){
    return p1[1] < p2[1];
  }
};

int main(){
  long long n,k;
  vector<vector<long long>> pr;

  cin >> n;

  for(int i=0;i<n;++i){
    long long a,b;
    cin >> a >> b;
    vector<long long> ab = {a,b};
    pr.push_back(ab);
  }

  cin >> k;
  
  sort(pr.begin(),pr.end(),COMPARSION());

  long counter = 0;

  for(vector<long long> i:pr){
    counter+=i[0];
    if(counter>=k){
      cout << i[1];
      break;
    }
  }

  return 0;
}

File: 9000-9999/9639 - Dinonun böyük massivi/solution.py
n = int(input())
list1 = []

for _ in range(n):
  a, b = map(int,input().split())
  list1.append([a, b])

k = int(input())

list1.sort(key = lambda x : x[1])

counter = 0
for i in list1:
  counter += i[0]
  if (counter>=k):
    print(i[1])
    break
